posthog/async_migrations/migrations/0002_fill_pdi2_from_pdi.py

```diff
@@ -0,0 +1,140 @@
+from constance import config
+from django.conf import settings
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.events import EVENTS_TABLE
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperation
+from posthog.constants import AnalyticsDBMS
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+from posthog.version_requirement import ServiceVersionRequirement
+
+
+class Migration(AsyncMigrationDefinition):
+
+    description = """
+        Schema change to migrate the data from the old person_distinct_id table
+        to the new person_distinct_id2 table. This is required as the existing
+        person_distinct_id does not use a strictly increasing id for
+        identifying the latest (distinct_id, person_id) pairing.
+
+        The issue is highlighted by the fact that we can have the scenario:
+
+            1. "person 1" is associated with "distinct_id A"
+            2. "person 1" is deleted resulting in a new row being written with is_deleted = 1
+            
+        The data can end up looking like:
+
+        _timestamp            |  distinct_id |  person_id  |  is_deleted
+        ----------------------+--------------+-------------+--------------
+        2019-01-01 10:10:10   |  A           |  1          |  0
+        2019-01-01 10:10:10   |  A           |  1          |  1
+
+        Hence there is no way of telling if the person was first deleted then
+        re-associated or the other way around. Hence at the moment with the old
+        table we're just checking that all rows relating to a (distinct_id,
+        person_id) pair are marked as is_deleted = 0.
+
+        The person_distinct_id table uses the `CollapsingMergeTree` engine, which
+        will at some point pair these two rows together, but we can't rely on
+        waiting. It also means that if we've have to be diligent with ensuring
+        we have written cancel rows correctly. It's good for cases where we
+        might want to perform aggregation over data, but `ReplacingMergeTree` is
+        a better fix where we just want the latest data.
+
+        The new schema includes a `version` column which should be strictly
+        increasing. At the time of writing, this version is a big int 
+        [updated in a
+        transaction](https://github.com/PostHog/posthog/blob/bcf2b6370f8d2205f1f7d5fb5f431124c3848691/plugin-server/src/worker/ingestion/properties-updater.ts#L58:L58)
+        within postgres, then propagated to clickhouse on successful commit via
+        kafka.
+
+        NOTE: on cloud we have already migrated team_id = 2, but adding
```

Comment:
Incorrect (we also have another team under it) and not really relevant context to be saved forever in this migration